modules/local/generate_summary.nf

```diff
@@ -0,0 +1,34 @@
+process GENERATE_SUMMARY {
+    tag "$meta.id"
+    label 'process_single'
+
+    if (params.enable_conda) {
+        exit 1, "Conda environments cannot be used when using the GENERATE_SUMMARY module. Please use docker or singularity containers."
+    }
+    container 'genomehubs/blobtoolkit:4.0.7'
+
+    input:
+    tuple val(meta), path(blobdir)
+    tuple val(meta), path(png)
```

Comment:
The png file isn't passed as an input of the command. Why is it listed as an input ?

Reply:
In the [generate_summary.smk](https://github.com/blobtoolkit/blobtoolkit/blob/main/src/blobtoolkit-pipeline/src/blobtoolkit-snakefiles/rules/generate_summary.smk) it is declared as input but the input in the shell command is only BlobDir: 

```
   rule generate_summary:
    """
    Use BlobTools2 filter to generate a dataset summary.
    """
    input:
        "{blobdir}/cumulative.png"
    output:
        "{blobdir}/summary.json"
    params:
        blobdir = lambda wc: wc.blobdir
    threads: 1
    log:
        "logs/{blobdir}/generate_summary.log"
    benchmark:
        "logs/{blobdir}/generate_summary.benchmark.txt"
    shell:
        """blobtools filter --summary {output} {params.blobdir} 2> {log}"""
```
But, I guess it can be removed since it is not used in the actual command.

Reply:
OK. I think it's a way of saying that this rule has to run _after_ the rule that generates `cumulative.png`, otherwise snakemake would run them in parallel (or in the opposite order). Here you're already enforcing the ordering in the sub-workflow itself, so no need to list it as input